hbase-server/src/main/java/org/apache/hadoop/hbase/io/hfile/HFileReaderImpl.java

```diff
@@ -554,8 +559,10 @@ protected int blockSeek(Cell key, boolean seekBefore) {
               + " path=" + reader.getPath());
         }
         offsetFromPos += Bytes.SIZEOF_LONG;
+        rowLen = ((blockBuffer.getByteAfterPosition(offsetFromPos) & 0xff) << 8)
+            ^ (blockBuffer.getByteAfterPosition(offsetFromPos + 1) & 0xff);
         blockBuffer.asSubByteBuffer(blockBuffer.position() + offsetFromPos, klen, pair);
-        bufBackedKeyOnlyKv.setKey(pair.getFirst(), pair.getSecond(), klen);
+        bufBackedKeyOnlyKv.setKey(pair.getFirst(), pair.getSecond(), klen, (short)rowLen);
```

Comment:
Whats the adv of decode rowLen here than in setKey()?  If the rowLen was already decoded for some other reason ya make sense to pass it than recalc later. But here am not fully sure

Reply:
IMO the rowLen is anyway getting decoded in the setKey method. And the same we can reuse and have it for rowLen. So I think it is better to have it decoded earlier. Because that rowLen decoding is happening per cell.

Reply:
I see how the usage is helpful.  While seek to a key within the seek while loop, we decode and keep the state of rowLen. Once seek is done and called getCell() there we dont have to decode again but reuse the saved state. make sense.  Though I have a comment.
Use BB#getShortAfterPosition() instead of having extra logic here.  that will be even perf wise better both in MBB and SBB

Reply:
`Use BB#getShortAfterPosition() instead of having extra logic here. that will be even perf wise better both in MBB and SBB`

You mean instead of getByteAfterPosition getting called 2 times just use getShortAfterPosition()?